.github/workflows/cnb.yml

```diff
@@ -0,0 +1,99 @@
+name: CNB
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '00 4 * * 1-5'
+
+defaults:
+  run:
+    # Setting an explicit bash shell ensures GitHub Actions enables pipefail mode too,
+    # rather than only error on exit (improving failure UX when pipes are used). See:
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell
+    shell: bash
+
+permissions:
+  actions: write
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-cnb:
+    name: Update .NET CNB version
+    runs-on: pub-hk-ubuntu-24.04-ip
+
+    steps:
+      - uses: actions/create-github-app-token@v1
+        id: generate-token
+        with:
+          app-id: ${{ vars.LINGUIST_GH_APP_ID }}
+          private-key: ${{ secrets.LINGUIST_GH_PRIVATE_KEY }}
+
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
```

Comment:
(optional) 

One way to make this easier to test via a branch workflow_dispatch is by adding an explicit `ref: main` here, similar to:
https://github.com/heroku/cnb-builder-images/blob/b9eb4f040ff0d3325508fc107cd382a2080d3bd7/.github/workflows/update-lifecycle.yml#L24-L25

Reply:
Good catch -- I've added the explicit `ref: main`. I didn't add it earlier so I could test the workflow (as a newer CNB release isn't available, and I didn't want to change the pinned CNB version on main) :)